components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/SystemDefaultClaimMetadataManager.java

```diff
@@ -0,0 +1,277 @@
+            if (LOCAL_CLAIM_DIALECT_URI.equals(entry.getKey())) {
+                continue;
+            }
+            List<Claim> externalClaims = entry.getValue().stream()
+                    .map(ExternalClaim.class::cast)
+                    .filter(claim -> localClaimURI.equals(claim.getMappedLocalClaim()))
+                    .map(Claim.class::cast)
+                    .collect(Collectors.toList());
+            mappedExternalClaims.addAll(externalClaims);
+        }
+        return mappedExternalClaims;
+    }
+
+    @Override
+    public boolean isMappedLocalClaim(String localClaimURI, int tenantId) throws ClaimMetadataException {
+
+        if (localClaimURI == null || StringUtils.isBlank(localClaimURI)) {
+            return false;
+        }
+
+        for (Map.Entry<String, List<Claim>> entry : claims.entrySet()) {
+            if (LOCAL_CLAIM_DIALECT_URI.equals(entry.getKey())) {
+                continue;
+            }
+            List<Claim> externalClaims = entry.getValue().stream()
+                    .map(ExternalClaim.class::cast)
+                    .filter(claim -> localClaimURI.equals(claim.getMappedLocalClaim()))
+                    .collect(Collectors.toList());
+            if (!externalClaims.isEmpty()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isLocalClaimMappedWithinDialect(String mappedLocalClaim, String externalClaimDialectURI, int tenantId) throws ClaimMetadataException {
+
+        if (externalClaimDialectURI == null || StringUtils.isBlank(externalClaimDialectURI)
+                || !claims.containsKey(externalClaimDialectURI)) {
+            return false;
+        }
+        if (mappedLocalClaim == null || StringUtils.isBlank(mappedLocalClaim)) {
+            return false;
+        }
+        return claims.get(externalClaimDialectURI).stream()
+                .filter(claim -> claim instanceof ExternalClaim)
+                .map(ExternalClaim.class::cast)
+                .anyMatch(claim -> mappedLocalClaim.equals(claim.getMappedLocalClaim()));
+    }
+
+    private static Map<String, String> filterClaimProperties(Map<String, String> claimProperties) {
+
+        claimProperties.remove(ClaimConstants.DIALECT_PROPERTY);
+        claimProperties.remove(ClaimConstants.CLAIM_URI_PROPERTY);
+        claimProperties.remove(ClaimConstants.ATTRIBUTE_ID_PROPERTY);
+
+        if (!claimProperties.containsKey(ClaimConstants.DISPLAY_NAME_PROPERTY)) {
+            claimProperties.put(ClaimConstants.DISPLAY_NAME_PROPERTY, "0");
+        }
+
+        if (claimProperties.containsKey(ClaimConstants.SUPPORTED_BY_DEFAULT_PROPERTY)) {
+            if (StringUtils.isBlank(claimProperties.get(ClaimConstants.SUPPORTED_BY_DEFAULT_PROPERTY))) {
+                claimProperties.put(ClaimConstants.SUPPORTED_BY_DEFAULT_PROPERTY, "true");
+            }
+        }
+
+        if (claimProperties.containsKey(ClaimConstants.READ_ONLY_PROPERTY)) {
```

Comment:
refer to previous comment

Reply:
Fixed with b159fa412a5dcb2d19480a24da4d91c582b5a501